pkg/controller/nodes/task/handler.go

```diff
@@ -598,6 +640,7 @@ func New(ctx context.Context, kubeClient executors.Client, client catalog.Client
 	return &Handler{
 		pluginRegistry: pluginMachinery.PluginRegistry(),
 		plugins:        make(map[pluginCore.TaskType]pluginCore.Plugin),
+		taskMetricsMap: make(map[pluginCore.TaskType]*taskMetrics),
```

Comment:
```suggestion
		taskMetricsMap: make(map[string]*taskMetrics),
```

or a new "MetricKey" type alias?